core/src/main/scala/org/apache/spark/io/CompressionCodec.scala

```diff
@@ -68,6 +69,7 @@ private[spark] object CompressionCodec {
     } catch {
       case e: ClassNotFoundException => None
       case e: IllegalArgumentException => None
+      case e: InvocationTargetException => None
```

Comment:
You can't do this -- this is a wrapper exception. I think you're just swallowing or masking an exception and it's not clear you can.